pandas/tests/io/test_excel.py

```diff
@@ -843,6 +843,13 @@ def test_read_excel_squeeze(self, ext):
         expected = pd.Series([1, 2, 3], name='a')
         tm.assert_series_equal(actual, expected)
 
+    def test_read_excel_engine_value(self, ext):
```

Comment:
can you parameterize with (for the ExcelFile), engine=['xlrd', 'openpyxl']

should fail for both

Reply:
Just to confirm my current implementation doesn't throw for the following case:
```
e = ExcelFile("blah.any", engine='openpyxl')
pd.read_excel(e, engine='openpyxl')
```
As around 15 of the test cases in `test_excel` used this construction - if you think the above should throw them I am happy to update the test cases - ( its just a simple update to the `cd_and_set_engine` fixture )